### PR TITLE
fix(web): prevent updating variants to have empty conditions (front-end only)

### PR DESCRIPTION
**Pull request by antonjoel82**
### What change does this PR introduce?

Front-end fixes for preventing empty variant conditions
- Introduces test to catch issue with saving variants with empty conditions
- Disable submit button and add tooltip when removing all conditions from a variant

### Why was this change needed?

Fixes NV-3147

### CODE REVIEW

File: apps/web/src/components/conditions/Conditions.tsx [hunk 1]
```diff
@@ -26,6 +26,19 @@ import { OnlineConditionRow } from './OnlineConditionRow';
...
+  isOpened: boolean;
+  isReadonly?: boolean;
+  onClose: () => void;
+  updateConditions: (data: IConditions[]) => void;
+  conditions?: IConditions[];
+  name: string;
+  label?: string;
+  filterPartsList: IFilterTypeList[];
+  defaultFilter?: FilterPartTypeEnum;
+  shouldDisallowEmptyConditions?: boolean;
```
Comment (antonjoel82) [inline apps/web/src/components/conditions/Conditions.tsx:39]: Flag here is optional / opt-in to preserve the original functionality for other callers of this component.

**Review by BiswaViraj — CHANGES_REQUESTED** [review]
@antonjoel82, Sorry for not adding this in the task description
But we will need to update the API too to disallow update of empty conditions variants.
[more details](https://linear.app/novu/issue/NV-3147/🔀-able-to-save-variants-without-any-conditions#comment-26036b30)

**Comment by antonjoel82** [comment]
> @antonjoel82, Sorry for not adding this in the task description But we will need to update the API too to disallow update of empty conditions variants. [more details](https://linear.app/novu/issue/NV-3147/%F0%9F%94%80-able-to-save-variants-without-any-conditions#comment-26036b30)

Sounds good, thanks for the call-out! I would typically do those in two different PRs, but am happy to do that in this PR as well if that's the preferred way.

**Comment by BiswaViraj** [comment]
@antonjoel82, it will be better to have two different PRs. I thought API changes were might be overlooked due to not being in the task description 🙏 🙈.
Also just how linear works, I'll suggest you to create a sub-issue for API over linear and tackle it on that task. This way we will have our linear and prs synced.
Let me know if you need anything.

**Comment by antonjoel82** [comment]
> @antonjoel82, it will be better to have two different PRs. I thought API changes were might be overlooked due to not being in the task description 🙏 🙈. Also just how linear works, I'll suggest you to create a sub-issue for API over linear and tackle it on that task. This way we will have our linear and prs synced. Let me know if you need anything.

Great, thank you! I'll update this PR description to make it more explicit, and have created a [sub-issue](https://linear.app/novu/issue/NV-3363/prevent-saving-variants-without-any-conditions-on-the-server) as requested.

**Review by BiswaViraj — APPROVED** [review]
🚀

**Review by LetItRock — APPROVED** [review]
awesome work!🔥 verified it with the dev testing and works like a charm :) let's now also implement the changes on API side :)

**Pull request merged**